query-engine/connectors/sql-query-connector/src/database/operations/write.rs

```diff
@@ -12,16 +16,82 @@ use std::{
 use tracing::log::trace;
 use user_facing_errors::query_engine::DatabaseConstraint;
 
+async fn generate_id(
+    conn: &dyn QueryExt,
+    conn_info: ConnectionInfo,
+    model: &ModelRef,
+    args: WriteArgs,
+    trace_id: Option<String>,
+) -> crate::Result<(Option<SelectionResult>, bool)> {
+    let pk = model.primary_identifier();
+
+    // fast path if it's not mysql
+    if conn_info.sql_family() != SqlFamily::Mysql {
+        return Ok((args.as_record_projection(pk.into()), false));
+    }
+
+    // Go to all the values and generate a select statement with the correct mysql function
+    let (pk_select, generated) = pk
+        .clone()
+        .selections()
+        .filter_map(|field| match field {
+            SelectedField::Scalar(x) if x.default_value.is_some() => x
+                .default_value
+                .clone()
+                .unwrap()
+                .dbgenerated_func()
+                .map(|func| (field.db_name().to_string(), func)),
+            _ => None,
+        })
+        .fold((Select::default(), false), |(query, generated), value| {
+            let alias = value.0;
+            let func = value.1.to_lowercase();
+            match func.trim() {
+                "(uuid())" => (query.value(native_uuid().alias(alias)), true),
```

Comment:
if a user does `(uuid( ))` will this match?

Reply:
You have to add a test to the above to prove it. Matching strings is always tricky.

Reply:
Nope, the rules for _this specific_ bug are clear and point out to the most common bug regarding dbgenerated (and they are not using spaces). I had an earlier version stripping spaces and all of that but it was a lot of complications for something that honestly, we will probably change in the future (I am still a strong believer the whole Id generation stuff should be rebuild). I see your point though, let me see how I can make you happy.